GVFS/GVFS.Common/GitStatusCacheConfig.cs

```diff
@@ -86,7 +86,7 @@ private static bool TryGetFromGitConfig(GitProcess git, string configName, int d
             error = string.Empty;
 
             GitProcess.Result result = git.GetFromConfig(configName);
-            if (result.HasErrors)
+            if (result.HasNonWarningErrors)
             {
                 if (result.Errors.Any())

```

Comment:
(This comment applies to change in `RetryConfig.cs` as well)

Rather than the current change, should this be updated to:

```
if (result.HasErrors)
{
    if (result.HasNonWarningErrors)
    {
        error = "Error while reading '" + configName + "' from config: " + result.Errors;
```

The current changes confused me a bit because based on this comment:

```
// Git returns non-zero for non-existent settings and errors.
```

This `if` block was intended to handle all non-zero return values from git.  With the new changes, this code block is only handling "fatal errors" and other errors (e.g. non-existing setting) are skipping the block and continuing on to the code that's meant to parse the output when git was able to successfully read the config value.

Reply:
> this code block is only handling "fatal errors" and other errors (e.g. non-existent setting) are skipping the block and continuing on to the code that's meant to parse the output when git was able to successfully read the config value

Yes and no. The very next statement after this if block tests for empty output, meaning use the default. Semantically I don't think anything changed but I agree it could be cleaner. I'm looking at other options.

Reply:
> Yes and no. The very next statement after this if block tests for empty output, meaning use the default. Semantically I don't think anything changed but I agree it could be cleaner. I'm looking at other options.

Yep that's a good point.  Maybe just remove (or move?) the `// Git returns non-zero for non-existent settings and errors.` comment?  I'm not sure that it's in the best place now.

Reply:
I simplified this codepath.